src/Features/CSharp/Portable/ConvertProgram/ConvertProgramTransform_ProgramMain.cs

```diff
@@ -84,11 +99,29 @@ public static async Task<Document> ConvertToProgramMainAsync(Document document,
             if (method.ParameterList.Parameters.Count == 1 && method.ParameterList.Parameters[0].Type is ArrayTypeSyntax arrayType)
                 method = method.ReplaceNode(arrayType.ElementType, PredefinedType(Token(SyntaxKind.StringKeyword)));
 
-            return FixupComments(generator.ClassDeclaration(
-                WellKnownMemberNames.TopLevelStatementsEntryPointTypeName,
-                accessibility: useDeclaredAccessibity ? programType.DeclaredAccessibility : Accessibility.NotApplicable,
-                modifiers: hasExistingPart ? DeclarationModifiers.Partial : DeclarationModifiers.None,
-                members: new[] { method }).WithLeadingTrivia(leadingTrivia));
+            ClassDeclarationSyntax? newClassDeclaration = null;
+
+            if (oldClassDeclaration is null)
+            {
+                // If we dodn't have any suitable class declaration in the same file then generate it
+                newClassDeclaration = FixupComments((ClassDeclarationSyntax)generator.ClassDeclaration(
+                    WellKnownMemberNames.TopLevelStatementsEntryPointTypeName,
+                    accessibility: useDeclaredAccessibity ? programType.DeclaredAccessibility : Accessibility.NotApplicable,
+                    modifiers: hasExistingPart ? DeclarationModifiers.Partial : DeclarationModifiers.None,
+                    members: new[] { method }).WithLeadingTrivia(leadingTrivia));
+            }
+            else
+            {
+                // Otherwise just add new member and process leading trivia
+
+                // Old class declaration is below top-level statements and is probably separated from them with a blank line.
+                // We want to remove this line to make class declaration begin from the first line of the file in the end
+                var oldTriviaWithoutBlankLines = oldClassDeclaration.GetLeadingTrivia().WithoutLeadingBlankLines();
+                newClassDeclaration = oldClassDeclaration.WithMembers(oldClassDeclaration.Members.Add(method))
+                                                         .WithLeadingTrivia(oldTriviaWithoutBlankLines.Union(leadingTrivia));
+            }
+
+            return newClassDeclaration;
```

Comment:
nit: no need for the `newClassDeclaration` local.  Can just return the items directly in the `if/else`.  that would be my preference.  but wasn't going to block on this.